lib/web3EventService.ts

```diff
@@ -364,8 +342,19 @@ export class Web3EventService {
       // invoke callback if there is one
       if (callback) {
         if (singly) {
-          log.forEach((e: DecodedLogEntryEvent<TEventArgs>) => { callback(error, e); });
+          for (const e of log) {
+            if (requiredDepth) {
+              await TransactionService.watchForConfirmedTransaction(e.transactionHash, null, requiredDepth);
+            }
+            callback(error, e);
+          }
         } else {
+
+          for (const e of log) {
```

Comment:
isn't this duplicate code of above?

Reply:
no, but it is redundant.  I've cleaned it up.  Good catch.